src/shared/utils/fs.ts

```diff
@@ -101,7 +101,7 @@ export async function findProjectFolders(logger: Logger, roots: string[], exclud
 	}));
 	const projectFoldersChecks = await Promise.all(projectFolderPromises);
 	const projectFolders = projectFoldersChecks
-		.filter((res) => res.exists)
+		.filter((res) => res.exists && !res.folder.includes(".dart_tool"))
```

Comment:
We should probably more this to the filter on line 94 (or an additional line next to it) to avoid the `hasPubspec` etc. checks on folders we'd then discard anyway.

We should also probably also put `${path.sep}` on each end just to ensure we only get folders with exactly that segment.